rasa/shared/core/flows/flow.py

```diff
@@ -399,6 +403,44 @@ def step_from_json(flow_step_config: Dict[Text, Any]) -> FlowStep:
         return BranchFlowStep.from_json(flow_step_config)
 
 
+def build_flow_steps_list(steps: List[Dict[Text, Any]]) -> List[FlowStep]:
+    flow_steps: List[FlowStep] = []
+
+    for i, step_config in enumerate(steps):
+        if step_config.get("next") is None:
+
+            if i == len(steps) - 1:
+                step_config["next"] = END_STEP
+            else:
+                step_config["next"] = steps[i + 1].get("id")
+
+        else:
+
+            if isinstance(step_config["next"], list):
+                updated_next_config = []
+
+                for link_config in step_config["next"]:
+                    if "then" in link_config and link_config.get("then") == "END":
+                        link_config["then"] = END_STEP
+
+                    if "else" in link_config and link_config.get("else") == "END":
```

Comment:
I think we do a lot of "double parsing" here: because the step isn't parsed into a dataclass already, we need to access the json properties here. Does it make sense to parse the steps first and then "link them together"?